### PR TITLE
1770: fixes for carousels

### PR DESCRIPTION
**Pull request by cableman**
* Fixed access callback for ajax loading ding lists in carousels
* Ensures that 0-hit search carousel don't make infinite loop in JS. 
* Fixes ratings in carousel items.

See https://platform.dandigbib.org/issues/1770

### CODE REVIEW

File: modules/p2/ding_list/include/functions.inc [hunk 1]
```diff
@@ -818,6 +818,11 @@ function ding_list_get_permission_permissions($permission = FALSE) {
 function ding_list_user_has_access($list, $permission = DING_LIST_PERMISSION_VIEW, $account = NULL, $token = FALSE) {
   $access = &drupal_static(__FUNCTION__, array());
 
+  // If used as menu access callback the list have to be loaded.
+  if (is_numeric($list)) {
+    $list = ding_list_get_list($list);
+  }
+
```
Comment (kasperg) [inline modules/p2/ding_list/include/functions.inc]: If we are to utilize the static cache then I would prefer to see this moved below the subsequent access check.

File: modules/p2/ding_list/include/functions.inc [hunk 2]
```diff
@@ -802,8 +802,8 @@ function ding_list_get_permission_permissions($permission = FALSE) {
 /**
  * Check if a user has access to a list, with the given permissions.
  *
- * @param object $list
- *   List entity.
+ * @param mixed $list
```
Comment (kasperg) [inline modules/p2/ding_list/include/functions.inc]: This should be `object|int`.

Comment (cableman) [reply modules/p2/ding_list/include/functions.inc]: fixed

File: modules/p2/ding_list/include/functions.inc (hunk 1 above)
Comment (cableman) [reply modules/p2/ding_list/include/functions.inc]: Nice catch. Was a very fast PR. :-) Fixed now

File: modules/ding_carousel/ding_carousel.theme.inc [hunk 3]
```diff
@@ -23,6 +23,17 @@ function template_preprocess_ding_carousel(&$vars) {
     );
   }
 
+  // This is an empty carousel, which slick don't handle well (as in it will
+  // make a infinite loop and kill the browser). So e.g. 0 hits search
+  // carousel's are fixed here be simply placing a single placeholder slide.
```
Comment (kasperg) [inline modules/ding_carousel/ding_carousel.theme.inc]: :+1: Appreciate the level of commenting here.

**Pull request merged**